applications/tari_console_wallet/src/init/mod.rs

```diff
@@ -77,6 +78,46 @@ pub enum WalletBoot {
     Recovery,
 }
 
+/// Get feedback, if available, for a weak passphrase
+fn get_password_feedback(passphrase: &SafePassword) -> Option<Vec<Suggestion>> {
+    if let Ok(passphrase_str) = std::str::from_utf8(passphrase.reveal()) {
+        match zxcvbn(passphrase_str, &[]) {
+            Ok(scoring) => scoring
+                .feedback()
+                .to_owned()
+                .map(|feedback| feedback.suggestions().to_owned()),
+            _ => None,
+        }
+    } else {
+        // Fail empty if we can't convert the passphrase back to a string, which shouldn't happen
+        None
+    }
```

Comment:
This `if`/`match` nesting feels non-idiomatic to me. Suggestions on how to improve it are welcome!

Reply:
Updated.